pkg/xds/envoy/clusters/cluster_builder.go

```diff
@@ -1,15 +1,15 @@
 package clusters
 
 import (
-	envoy_api "github.com/envoyproxy/go-control-plane/envoy/api/v2"
-)
+	"errors"
 
-// ClusterConfigurer is responsible for configuring a single aspect of the entire Envoy cluster,
-// such as filter chain, transparent proxying, etc.
-type ClusterConfigurer interface {
-	// Configure configures a single aspect on a given Envoy cluster.
-	Configure(cluster *envoy_api.Cluster) error
-}
+	envoy_api_v2 "github.com/envoyproxy/go-control-plane/envoy/api/v2"
+	envoy_api "github.com/envoyproxy/go-control-plane/envoy/config/cluster/v3"
```

Comment:
This seems inconsistent with the below `v2` and `v3` pointing out.

Shall e just use `envoy_api_v3`? I don't mind either way, but maybe being explicit will make supporting these through thime slightly easier.